model_compression_toolkit/quantizers_infrastructure/keras/quantize_wrapper.py

```diff
@@ -57,51 +52,104 @@ def _weight_name(name: str) -> str:
     class KerasQuantizationWrapper(tf.keras.layers.Wrapper):
         def __init__(self,
                      layer,
-                     dispatcher: NodeQuantizationDispatcher,
+                     weight_quantizers: Dict[str, BaseInferableQuantizer] = None,
+                     activation_quantizers: List[BaseInferableQuantizer] = None,
                      **kwargs):
             """
-            Keras Quantization Wrapper takes a keras layer and dispatcher and infer a quantized layer.
+            Keras Quantization Wrapper takes a keras layer and quantizers infer a quantized layer.
 
             Args:
                 layer: A keras layer.
-                dispatcher: A node quantization dispatcher.
+                weight_quantizers: A dictionary between a weight's name to its quantizer.
+                activation_quantizers: A list of activations quantization, one for each layer output.
             """
             super(KerasQuantizationWrapper, self).__init__(layer, **kwargs)
             self._track_trackable(layer, name='layer')
-            self._dispatcher = dispatcher
+            self.weight_quantizers = weight_quantizers if weight_quantizers is not None else dict()
+            self.activation_quantizers = activation_quantizers if activation_quantizers is not None else list()
+
+        def add_weight_quantizer(self, param_name: str, quantizer: BaseInferableQuantizer):
+            """
+            This function adds a weight quantizer to existing wrapper
+
+            Args:
+                param_name: The name of the parameter to quantize
+                quantizer: A quantizer.
+
+            Returns: None
+
+            """
+            self.weight_quantizers.update({param_name: quantizer})
+
+        @property
+        def is_activation_quantization(self) -> bool:
+            """
+            This function check activation quantizer exists in wrapper.
+            Returns: a boolean if activation quantizer exists
+
+            """
+            return self.num_act_quantizers > 0
+
+        @property
+        def is_weights_quantization(self) -> bool:
```

Comment:
Be consistent with the term "weights" vs "weight" to prevent confusion and future syntax errors.
I personally prefer "weights", which is more common in the rest of our code (I don't know why in the dispatcher most of the methods use "weight", I would have change all of them but it's up to you)

Reply:
it should be weights